consensus/state_processing/src/per_block_processing.rs

```diff
@@ -575,6 +615,14 @@ pub fn process_withdrawals<T: EthSpec, Payload: AbstractExecPayload<T>>(
                 )?;
             }
 
+            if let Ok(mut pending_partial_withdrawals) = state
+                .pending_partial_withdrawals_mut()
+                .map(|ppw| Vec::from(std::mem::replace(ppw, Vec::new().into())))
+            {
+                pending_partial_withdrawals.drain(0..partial_withdrawals_count);
```

Comment:
This could be quite expensive with tree states done naively. It's possible to do a specialized operation on milhouse that splits the tree at some index while potentially preserving part of the hash cache @michaelsproul 

@ethDreamer can you add a `TODO(EIP7251)` to not forget?